src/Microsoft.Framework.CopyOnWriteDictionary.Internal/CopyOnWriteDictionary.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Internal
```

Comment:
won't the `internal` -> `public` change cause a bevy of downstream issues as projects build, especially when one project depends on multiple projects that use Common packages?  why not stick with `internal` for build-time dependencies?  this might even work around aspnet/XRE#1237 -- don't need transitive shared source dependencies but do need to keep use of shared classes `internal`.